### ISSUE TITLE
[Enhancement] - Trying 3rd Eye

### ISSUE THREAD

**Issue by Prime-Script** [DanielCoelh0/qb-lift#1]
Hi, I have to the elevator to work with 3rd eye until you come to using the menu, when you open the menu for the elevator the buttons appear to be doing nothing... I have found if you restart the script while in the menu it will let you use the menu for each button, and take you to the floor (Can I get any advice)?

**Comment by DanielCoelh0** [comment]
> Hi, I have to the elevator to work with 3rd eye until you come to using the menu, when you open the menu for the elevator the buttons appear to be doing nothing... I have found if you restart the script while in the menu it will let you use the menu for each button, and take you to the floor (Can I get any advice)?

To confirm, if I understand correctly, when you use the 3rd eye to show the menu of floors, when you select the floor, it does nothing? 
The server / client console doesn't show any errors?

**Comment by Prime-Script** [comment]
> > Hi, I have to the elevator to work with 3rd eye until you come to using the menu, when you open the menu for the elevator the buttons appear to be doing nothing... I have found if you restart the script while in the menu it will let you use the menu for each button, and take you to the floor (Can I get any advice)?
> 
> To confirm, if I understand correctly, when you use the 3rd eye to show the menu of floors, when you select the floor, it does nothing? The server / client console doesn't show any errors?

Yes correct, the qb-menu shows, but when you click (e.g. First Floor does nothing) but if I'm in the menu and restart the script, then click the First Floor the progress bar starts... (No console / f8 errors)

**Comment by Prime-Script** [comment]
![image](https://user-images.githubusercontent.com/93712286/152656606-c7c39ebb-dc63-4d61-817a-10376181481b.png)
So at this stage the menu does nothing, until you restart the script while staying in the menu, then it works...

**Comment by DanielCoelh0** [comment]
I understand, it works by default, but if it doesn't give errors in the f8 console, it may be that something is missing in the code when adapting to the 3rd eye

**Comment by Prime-Script** [comment]
> I understand, it works by default, but if it doesn't give errors in the f8 console, it may be that something is missing in the code when adapting to the 3rd eye

The only section of the script that I changed was make the Thread For the Marker a Net Event so I can call it in the qb-target? When I add the Marker back in it works.

**Comment by DanielCoelh0** [comment]
The function "changeFloor(data)" is responsible for making the player change floor. If it does not teleport, it is because some value is passing null.

**Comment by Prime-Script** [comment]
> The function "changeFloor(data)" is responsible for making the player change floor. If it does not teleport, it is because some value is passing null.

I added prints into the script to see if the button where even triggering, and I got nothing at all. Could I send you the version I have and would you be able to look at it?

**Comment by DanielCoelh0** [comment]
Yes. You can send to my discord. kumapt#0001 if you want
